app/org/sagebionetworks/bridge/models/schedules/SchedulerFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import org.sagebionetworks.bridge.validators.ScheduleValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+public class SchedulerFactory {
+
+    public static TaskScheduler getScheduler(String schedulePlanGuid, Schedule schedule) {
+        // TODO: This validation will be moved to create/update, not the time of use.
+        Validate.entityThrowingException(new ScheduleValidator(), schedule);
+        
+        if (schedule.getCronTrigger() != null) {
+            return new CronTaskScheduler(schedulePlanGuid, schedule);
+        }
+        return new IntervalTaskScheduler(schedulePlanGuid, schedule);
```

Comment:
Would it be cleaner for TaskScheduler to be an immutable singleton that takes in the guid and schedule as an input, rather than creating a new one for each schedule? At the very least, this saves you the trouble of creating a scheduler cache.